compiler/sempass2.nim

```diff
@@ -577,6 +577,9 @@ proc trackOperand(tracked: PEffects, n: PNode, paramType: PType) =
         markGcUnsafe(tracked, a)
       elif tfNoSideEffect notin op.flags:
         markSideEffect(tracked, a)
+  if paramType != nil and paramType.kind == tyVar:
```

Comment:
There is such a check in `makeVolatile` itself. I didn't see a reason for duplicating it and the rest of the code is following the same pattern.